components/match2/wikis/halo/match_group_input_custom.lua

```diff
@@ -532,23 +524,9 @@ end
 --
 -- opponent related functions
 --
-function opponentFunctions.getTeamNameAndIcon(template, date)
-	local team, icon
-	date = mw.getContentLanguage():formatDate('Y-m-d', date or '')
-	template = (template or ''):lower():gsub('_', ' ')
-	if template ~= '' and template ~= 'noteam' and
-		mw.ext.TeamTemplate.teamexists(template) then
-
-		local templateData = mw.ext.TeamTemplate.raw(template, date)
-		icon = templateData.image
-		if icon == '' then
-			icon = templateData.legacyimage
-		end
-		team = templateData.page
-		template = templateData.templatename or template
-	end
-
-	return team, icon, template
+function opponentFunctions.getIcon(template)
+	local raw = mw.ext.TeamTemplate.raw(template)
+	return raw and Logic.emptyOr(raw.image, raw.legacyimage)
```

Comment:
Date has been removed from the template call, meaning historic images won't work (applies for all wiki customs obv)

Reply:
The template passed in has already been resolved to a historical template. So passing date to `mw.ext.TeamTemplate.raw` is not needed.